### PR TITLE
Move to AggregateFunctionExpr

### PR DESCRIPTION
**Pull request by ameyc**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -55,4 +55,4 @@ serde_json = "1"
 base64 = "0.22.1"
 chrono = { version = "0.4.38", default-features = false }
 itertools = "0.13"
-pyo3 = { version = "0.21.2", features = ["experimental-async"] }
+pyo3 = { version = "0.22.2", features = ["experimental-async"] }
```
Comment (emgeee) [inline Cargo.toml:58]: did this dep need to be updated? I think datafusion-python currently requires pyo3 0.21

File: crates/core/src/physical_plan/continuous/streaming_window.rs [hunk 2]
```diff
@@ -225,7 +227,7 @@ impl StreamingWindowExec {
             &input.schema(),
             group_by.expr(),
             &aggr_expr,
-            group_by.contains_null(),
+            false, //group_by.contains_null(),
```
Comment (emgeee) [inline crates/core/src/physical_plan/continuous/streaming_window.rs:230]: what's up with this comment?

File: py-denormalized/Cargo.toml [hunk 3]
```diff
@@ -10,7 +10,7 @@ name = "denormalized_python"
 crate-type = ["cdylib"]
 
 [dependencies]
-pyo3 = { version = "0.21.2", features = ["experimental-async"] }
+pyo3 = { workspace = true }
```
Comment (emgeee) [inline py-denormalized/Cargo.toml:13]: did we need pyo3 out of the py-denormalized crate?

Comment (ameyc) [reply py-denormalized/Cargo.toml:13]: yes it did. `cargo update` doesn't work otherwise because it goes down a circular resolver path.

**Pull request merged**